Source/WebCore/dom/Document.cpp

```diff
@@ -5470,10 +5471,23 @@ bool Document::canAcceptChild(const Node& newChild, const Node* refChild, Accept
     return true;
 }
 
+ClonedDocumentType Document::clonedDocumentType() const
+{
+    if (isXMLDocument()) {
+        if (isXHTMLDocument())
+            return ClonedDocumentType::XMLDocument;
+        return ClonedDocumentType::XMLDocument;
```

Comment:
If these both return XMLDocument, why have the branch for isXHTMLDocument? (Edit: it seems like this is a bug, as below we create a type for XHTML as well.)

Reply:
XHTML was an oversight.  Thanks!